### PR TITLE
Fix #741: navigation bar moves to right on click of heart icon

### LINKED ISSUE #741: 🐛 [Bug report]: Glitch in homepage navigation bar
### Descrcibe the bug?

In the homepage, while clicking on the heart icon in the navigation bar, the navigation bar moves right.

### Steps to reproduce the bug?

1. Go to `http://localhost:3000` or `https://reactplay.io/` on laptop/desktop browser.
2. Click on `heart icon` on the top right.
3. You will find that navigation bar is moving to the right.

### Expected behavior

Navigation bar shouldn't change position because of clicking actions.

### Desktop (Please provide your system information)

OS: MacOS
Browser: Chrome
Version: 106.0.5249.119

### Mobile (Please provide your device information)

_No response_

### Screenshot / Screenshare

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kd100100**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

In the homepage, while clicking on the heart icon in the navigation bar, the navigation bar moves right.

Fixes #741 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tested in local environment in cypress.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Sachin-chaurasiya** [comment]
Hello @kd100100 , Thanks for the PR, Can you attach some screenshots or screen recordings?

**Comment by kd100100** [comment]
[nav-error-recording.webm](https://user-images.githubusercontent.com/64349912/197127493-e512ee07-4a53-446e-b72d-2e1538eb2db6.webm)

**Comment by kd100100** [comment]
@atapas @koustov Could you please review this?

**Comment by Sachin-chaurasiya** [comment]
> > Hello @kd100100 , Thanks for the PR, Can you attach some screenshots or screen recordings?
> 
>  nav-error-recording.webm

@kd100100 , Can you share the recording for the fix?

**Comment by kd100100** [comment]
> > > Hello @kd100100 , Thanks for the PR, Can you attach some screenshots or screen recordings?
> > 
> > 
> > nav-error-recording.webm
> 
> @kd100100 , Can you share the recording for the fix?

@Sachin-chaurasiya Here is the recording of the fix.

[fixed-nav-error-recording.webm](https://user-images.githubusercontent.com/64349912/197982336-d4196ed6-e143-4307-af31-2e478a1b0137.webm)

**Comment by Sachin-chaurasiya** [comment]
> > > > Hello @kd100100 , Thanks for the PR, Can you attach some screenshots or screen recordings?
> > > 
> > > 
> > > nav-error-recording.webm
> > 
> > 
> > @kd100100 , Can you share the recording for the fix?
> 
> @Sachin-chaurasiya Here is the recording of the fix.
> 
>  fixed-nav-error-recording.webm

Thanks @kd100100

File: src/common/header/header.css [hunk 1]
```diff
@@ -20,10 +20,6 @@
   }
 }
 
-.app-header-home {
-  position: fixed;
-}
-
```
Comment (Sachin-chaurasiya) [inline src/common/header/header.css:26]: @kd100100 , If we remove this class then the header will be no longer fixed. is this change intentional? is yes then why?

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
I have verified on the live site and it's working fine. can you double-check is this issue exists?

https://github.com/reactplay/react-play/pull/742#pullrequestreview-1156912425

File: src/common/header/header.css (hunk 1 above)
Comment (kd100100) [reply src/common/header/header.css:26]: > I have verified on the live site and it's working fine. can you double-check is this issue exists?
> 
>  Screen.Recording.2022-10-26.at.10.07.07.PM.mov

@Sachin-chaurasiya, I have checked this issue on (MacOS - Chrome, Safari, Firefox and Windows - Chrome) now. This issue still exists on the live site `https://reactplay.io/`. In your recording I don't see a scrollbar, I guess you are checking on tablet. Could you please try on laptop/desktop?

Comment (kd100100) [reply src/common/header/header.css:26]: > @kd100100 , If we remove this class then the header will be no longer fixed. is this change intentional? is yes then why?

### Cause of the problem
When the heart is clicked the popup turns homepage into a non-scrollable part. So, the scrollbar disappears and since it was in `fixed` position, it expands itself to screen's width. That is what is causing the alignment changes in navigation bar.

### Solution
When the navigation bar is in `block` instead of `fixed` position like the other pages are currently, this problem doesn't seem to appear. I have attached the screen recoding of the behaviour change with inspect panel.

https://user-images.githubusercontent.com/64349912/198178871-53dc86a7-1621-420f-a272-8569bb7eb1e9.mp4

### Possible change that can be implemented
When position `fixed` is removed, the webpage goes down by 60px. We can fix it using margin.

Comment (kd100100) [reply src/common/header/header.css:26]: > @kd100100 , If we remove this class then the header will be no longer fixed. is this change intentional? is yes then why?

@Sachin-chaurasiya 

### Cause of the problem
When the heart is clicked the popup turns homepage into a non-scrollable part. So, the scrollbar disappears and since it was in `fixed` position, it expands itself to screen's width. That is what is causing the alignment changes in navigation bar.

### Solution
When the navigation bar is in `block` instead of `fixed` position like the other pages are currently, this problem doesn't seem to appear. I have attached the screen recoding of the behaviour change with inspect panel.

https://user-images.githubusercontent.com/64349912/198178871-53dc86a7-1621-420f-a272-8569bb7eb1e9.mp4

### Possible change that can be implemented
When position `fixed` is removed, the webpage goes down by 60px. We can fix it using margin.

**Comment by kd100100** [comment]
> I have verified on the live site and it's working fine. can you double-check is this issue exists?
> 
>  Screen.Recording.2022-10-26.at.10.07.07.PM.mov

@Sachin-chaurasiya Just found out that this issue occurs when connected to external mouse. Please check connecting to an external mouse.

File: src/common/header/header.css (hunk 1 above)
Comment (kd100100) [reply src/common/header/header.css:26]: > I have verified on the live site and it's working fine. can you double-check is this issue exists?
> 
>  Screen.Recording.2022-10-26.at.10.07.07.PM.mov

@Sachin-chaurasiya Just found out that this issue occurs when connected to external mouse. Please check connecting to an external mouse.

Comment (Sachin-chaurasiya) [reply src/common/header/header.css:26]: > https://react-play-git-fork-kd100100-nav-glitch-reactplayio.vercel.app/

Hello @kd100100 , I checked and it's working fine.

Comment (Sachin-chaurasiya) [reply src/common/header/header.css:26]: > > @kd100100 , If we remove this class then the header will be no longer fixed. is this change intentional? is yes then why?
> 
> @Sachin-chaurasiya
> 
> ### Cause of the problem
> When the heart is clicked the popup turns homepage into a non-scrollable part. So, the scrollbar disappears and since it was in `fixed` position, it expands itself to screen's width. That is what is causing the alignment changes in navigation bar.
> 
> ### Solution
> When the navigation bar is in `block` instead of `fixed` position like the other pages are currently, this problem doesn't seem to appear. I have attached the screen recoding of the behaviour change with inspect panel.
> 
>  nav-glitch-inspect.mp4 
> ### Possible change that can be implemented
> When position `fixed` is removed, the webpage goes down by 60px. We can fix it using margin.

@kd100100 , We want the header to be fixed and if you check on the live site we have fixed the header.

Comment (kd100100) [reply src/common/header/header.css:26]: @Sachin-chaurasiya This issue occurs when connected to external mouse. Could you connect an external mouse and see if this issue is resolved.

Just wanted to know, why do you want it `position fixed` only on the home page?

Comment (Sachin-chaurasiya) [reply src/common/header/header.css:26]: @kd100100 , Yes I tested with an external mouse and it's working fine.

@kd100100 , We have a header fixed on every page, please go through all the pages.

Comment (kd100100) [reply src/common/header/header.css:26]: @Sachin-chaurasiya I checked on all pages. It has `position: relative`. The following image is plays page.

Screenshot 2022-10-27 at 1 32 32 PM

Comment (Sachin-chaurasiya) [reply src/common/header/header.css:26]: @kd100100 , Ah, got it.
But still we want header to be fixed on home page as well.

Comment (Sachin-chaurasiya) [reply src/common/header/header.css:26]: I checked the preview https://react-play-ard3746fr-reactplayio.vercel.app/ and the header is not fixed. we can't break another thing to fix one thing.

**Comment by Sachin-chaurasiya** [comment]
@kd100100 , Can you address the comments?

File: src/common/header/header.css (hunk 1 above)
Comment (kd100100) [reply src/common/header/header.css:26]: > I checked the preview https://react-play-ard3746fr-reactplayio.vercel.app/ and the header is not fixed. we can't break another thing to fix one thing.

I been trying to come up with a solution but not able to find. If you have one, could you suggest me?

Comment (Sachin-chaurasiya) [reply src/common/header/header.css:26]: @kd100100 , I have verified it and it's not reproducible even with external mouse.

Comment (kd100100) [reply src/common/header/header.css:26]: Hi @Sachin-chaurasiya. Sorry for the late reply.
I have checked again and the problem still exists in my browser (`https://reactplay.io/`). But I am not sure if it has to do something with the browser settings. Shall we close this PR without merging?

Comment (Sachin-chaurasiya) [reply src/common/header/header.css:26]: Thanks for confirmation @kd100100, I will close this PR, and you can work on other issues.
Happy coding!

**Comment by Sachin-chaurasiya** [comment]
Closing this PR as the issue is not reproducible.
